tests/i18n_test.cpp

```diff
@@ -0,0 +1,258 @@
+
+   I18n::load_language_file("fr", "French", "data/languages/tests/fr.txt");
+   BOOST_CHECK_EQUAL("fr", I18n::get_locale());
+   BOOST_CHECK_EQUAL("French", I18n::get_language_name());
+
+   I18n::load_language_file("it", "Italian", "data/languages/tests/it.txt");
+   BOOST_CHECK_EQUAL("it", I18n::get_locale());
+   BOOST_CHECK_EQUAL("Italian", I18n::get_language_name());
+
+}
+
+
+
+
+BOOST_FIXTURE_TEST_CASE(returns_false_when_unable_to_load_a_language_file, Fixture)
+{
+   BOOST_CHECK_EQUAL(false, I18n::load_language_file("foo", "Fooanese", "foo/folder/file_that_doesnt_exist.txt"));
+}
+
+
+
+
+BOOST_FIXTURE_TEST_CASE(successfully_populates_labels_when_loading_a_language_file, Fixture)
+{
+   BOOST_REQUIRE_EQUAL(true, I18n::load_language_file("en", "English", "data/languages/tests/en.txt"));
+
+   BOOST_CHECK_EQUAL("Welcome to Allegro!", I18n::t("welcome_message"));
+   BOOST_CHECK_EQUAL("start", I18n::t("start_button"));
+}
+
+
+
+
+BOOST_FIXTURE_TEST_CASE(can_set_a_locale, Fixture)
+{
+   I18n::set_languages_folder(TEST_FOLDER);
+
+   BOOST_CHECK_EQUAL(true, I18n::set_locale("en"));
+   BOOST_CHECK_EQUAL(true, I18n::set_locale("fr"));
+   BOOST_CHECK_EQUAL(true, I18n::set_locale("it"));
+}
+
+
+
+
+BOOST_FIXTURE_TEST_CASE(returns_false_when_unable_to_set_a_locale, Fixture)
+{
+   I18n::set_languages_folder(TEST_FOLDER);
+
+   BOOST_CHECK_EQUAL(false, I18n::set_locale("not_a_locale"));
+}
+
+
+
+
+BOOST_FIXTURE_TEST_CASE(can_retrieve_the_current_locale, Fixture)
+{
+   I18n::set_languages_folder(TEST_FOLDER);
+
+   I18n::set_locale("en");
+   BOOST_CHECK_EQUAL("en", I18n::get_locale());
+
+   I18n::set_locale("fr");
+   BOOST_CHECK_EQUAL("fr", I18n::get_locale());
+
+   I18n::set_locale("it");
+   BOOST_CHECK_EQUAL("it", I18n::get_locale());
+}
+
+
+
+
+BOOST_FIXTURE_TEST_CASE(loads_translations_when_a_locale_is_set, Fixture)
+{
+   I18n::set_languages_folder(TEST_FOLDER);
+
+   I18n::set_locale("en");
+   BOOST_CHECK_EQUAL("Welcome to Allegro!", I18n::t("welcome_message"));
+   BOOST_CHECK_EQUAL("start", I18n::t("start_button"));
+
+   I18n::set_locale("fr");
+   BOOST_CHECK_EQUAL("Bienvenue à Allegro!", I18n::t("welcome_message"));
+   BOOST_CHECK_EQUAL("démarrer", I18n::t("start_button"));
+
+   I18n::set_locale("it");
+   BOOST_CHECK_EQUAL("Benvenuti a Allegro!", I18n::t("welcome_message"));
+   BOOST_CHECK_EQUAL("inizio", I18n::t("start_button"));
+}
+
+
+
+
+BOOST_FIXTURE_TEST_CASE(returns_an_empty_string_when_a_translation_does_not_exist_for_that_label, Fixture)
+{
+   BOOST_REQUIRE_EQUAL(true, I18n::load_language_file("en", "English", "data/languages/tests/en.txt"));
+   BOOST_CHECK_EQUAL("", I18n::t("label_that_does_not_exist"));
+}
+
+
+
+
+BOOST_FIXTURE_TEST_CASE(formats_a_translation_with_variable_args, Fixture)
+{
+   I18n::set_languages_folder(TEST_FOLDER);
+
+   I18n::set_locale("en");
+   BOOST_CHECK_EQUAL("Travel time is 6 hours and 45 minutes.", I18n::ft("travel_time", 6, 45));
```

Comment:
should this be `tf(...)` rather than `ft(...)`?  It is modeled after the function `printf()` after all.